src/components/Store/PeopleStore.tsx

```
@@ -0,0 +1,58 @@
+import React, { useReducer } from 'react';
+import { Person } from '../../types';
+
+interface State {
+  people: Person[];
+  loading: boolean;
+  error: boolean;
+  loaded: boolean;
+}
+
+type Action =
+  | { type: 'FETCH_START' }
+  | { type: 'FETCH_SUCCESS'; payload: Person[] }
+  | { type: 'FETCH_ERROR' };
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case 'FETCH_START':
+      return { ...state, loading: true, error: false, loaded: false };
+    case 'FETCH_SUCCESS':
+      return {
+        people: action.payload,
+        loading: false,
+        error: false,
+        loaded: true,
+      };
+    case 'FETCH_ERROR':
+      return { ...state, loading: false, error: true };
+    default:
+      return state;
+  }
+}
+
+const initialState: State = {
+  people: [],
+  loading: false,
+  error: false,
+  loaded: false,
+};
+
+type Props = {
+  children: React.ReactNode;
+};
+
+export const StateContext = React.createContext(initialState);
+export const DispatchContext = React.createContext<React.Dispatch<Action>>(
```

`DispatchContext` is created with a default `() => {}`. A no-op default can hide errors and the function signature may not match `React.Dispatch<Action>` exactly. Consider one of these safer patterns:
- Use `createContext<React.Dispatch<Action> | undefined>(undefined)` and assert/check the value after `useContext`, or
- Provide a default that throws (e.g. `() => { throw new Error('Dispatch not initialized') }`) so accidental use outside provider fails loudly.
This makes misuse easier to detect and keeps typing safer.